src/util/address.rs

```diff
@@ -300,6 +300,27 @@ impl Address {
         self.address_type().is_some()
     }
 
+    /// Get an [Address] from an output script (scriptPubkey).
+    pub fn from_script(script: &script::Script, network: Network) -> Option<Address> {
```

Comment:
You mean with `Into` and `From`? I don't think `addr.payload.into()` is good UX, so there'd still be `to_script_pubkey()`, right? So then it'd just be for code location? So that From and Into could be neatly defined underneath each other?

Reply:
@stevenroose It doesn't need to be `Into` and `From`, I'm just saying that there seems to be a clear conversion between `Payload` and `Script` regardless of the additional information in `Address`, so we should do that and use that conversion in our `Address` code.

Reply:
Addressed this.